### ISSUE TITLE
Converting std::process::Child::id() to struct Pid requires a cast

### ISSUE THREAD

**Issue by marmistrz** [nix-rust/nix#656]
Currently we need something like:
```
let child = Command::new("ls").arg("-l").spawn().unwrap();
let pid = Pid::from_raw(child.id() as i32);
```
while `id()` returns a `u32`.
```
fn id(&self) -> u32
```
This itself poses no risk on Linux, since the hard PID limit is 2^22, but this cast should not be necessary, imho.

Btw. `nix::unistd::Pid` is not mentioned in the documentation.

**Comment by Susurrus** [comment]
`i32` should be correct, this looks like a bug with `.id()` in stdlib. I'd suggest bringing it up there. I'd love to reduce friction with the stdlib, but it doesn't make sense that they'd return a `u32` while the underlying `pid_t` type on POSIX is i32. Maybe Windows only deals with `u32`s, so they decided to do that.

Regarding the lack of docs, `Pid` does have docs in the source. It's not been in a released version, so `docs.rs` is out of date, but you can generate the docs locally like `cargo doc --no-deps --open`.

**Comment by marmistrz** [comment]
Yep, in glibc it's a singed integer: https://www.gnu.org/software/libc/manual/html_node/Process-Identification.html

**Comment by Susurrus** [comment]
@marmistrz Thanks for raising that with upstream, let's sit on this for a day or so and see where that discussion goes.

**Comment by marmistrz** [comment]
@Susurrus upstream won't change it. Will nix?

**Comment by marmistrz** [comment]
What about:
```
pub fn<T: Int> from_raw(pid: T) -> Self;
```

**Comment by Susurrus** [comment]
There's already `Pid::from_raw(pid_t)`. So you're suggesting adding the above as well? I think it'd need to be unsafe or return a `Result` for it to really be feasible.

**Comment by frangio** [comment]
What about a `ChildExt` trait implemented for `std::process::Child` that adds a `pid(&self) -> Pid` method?

**Comment by SteveLauC** [comment]
The stdlib uses `u32` because [it is `u32` on Windows](https://github.com/rust-lang/rust/pull/43140/files#diff-28a2ae73555927a50a1fcd528e24e27b52e2dd0445df7011c25789d62905ff4c). On POSIX platforms, even though the stdlib uses `u32` , you can still safely cast it because it is guaranteed to be in the scope of `[0, i32::MAX]`. So `Pid::from_raw(Child::id() as i32)` would totally work.

Honestly I don't think we should introduce new traits or other helper functions for this safe type cast
